### PR TITLE
Additional Args and configurable docker image source (RDT-971)

### PR DESCRIPTION
**Pull request by timoxd7**
## Configurable Docker image source

Added `esp_idf_root` to be used as source for the docker image. Can be used to select docker images hosted on different accounts or servers.

## Additional arguments

Added `additional_args` to add custom arguments to the docker run command. Can be used to enable ccache usage, only add ccache ci action and add the following lines:

    -v ./.ccache:/root/.ccache -e CCACHE_DIR=/root/.ccache

Only make sure the `.ccache` folder is created if no cache was already build (always on the first run).

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -12,6 +12,14 @@ inputs:
     description: "Version of ESP-IDF docker image to use"
     default: "latest"
     required: false
+  esp_idf_root:
```
Comment (kumekay) [inline action.yml]:
```suggestion
  esp_idf_docker_image:
```

File: action.yml [hunk 2]
```diff
@@ -12,6 +12,14 @@ inputs:
...
+    description: "Docker Repository to obtain the ESP-IDF image from"
+    default: "espressif/idf"
+    required: false
+  additional_args:
```
Comment (kumekay) [inline action.yml]:
```suggestion
  extra_docker_args:
```

**Review by kumekay — COMMENTED** [review]
@timoxd7 thank you for the contribution, it looks good to me.

I would only suggest a bit more precise names for the parameters you introduced.

And could you please describe them in the README.md too?

**Comment by key** [comment]
It would be good if the image source could be switched.
@timoxd7 does not watch this PR?

**Comment by timoxd7** [comment]
@key Sorry, forgot about it. Will check right now!

File: action.yml (hunk 1 above)
Comment (timoxd7) [reply action.yml]: Done

File: action.yml (hunk 2 above)
Comment (timoxd7) [reply action.yml]: Done

**Comment by timoxd7** [comment]
@kumekay I just changed the names and added a documentation to the README.md file

**Comment by kumekay** [comment]
@timoxd7 Thank you, looks and works good,
I've tried it here https://github.com/kumekay/test_idf_action/actions/runs/11616606397

**Pull request merged**